packages/myst-cli/src/transforms/code.ts

```diff
@@ -194,3 +194,45 @@ export function propagateBlockDataToCode(session: ISession, vfile: VFile, mdast:
     if (outputNode && !outputNode.visibility) outputNode.visibility = 'show';
   });
 }
+
+/**
+ * Because jupytext style notebooks can introduce code-cells in blocks
+ * after explicit thematic breaks, we need to lift them up to the parent
+ * to ensure that all executable code blocks are at the top level of
+ * the document.
+ *
+ * @param mdast
+ */
+export function transformLiftCodeBlocksInJupytext(mdast: GenericParent) {
+  const flattened = (mdast.children as GenericParent[]).reduce((acc, node) => {
+    if (node.type !== 'block' || !node.children) return [...acc, node];
+
+    const buriedCodeCells = selectAll('block:has(block)', node);
+    if (buriedCodeCells.length === 0) return [...acc, node];
```

Comment:
This would basically happen anyway even without the early return, right? It would just go in a big circle reconstructing the block, ending up equivalent to how it started.

Reply:
If we did that though, we would lose block metadata!

I guess as-is, we do lose any metadata on the top-level blocks (though the code/output blocks should keep their metadata since those blocks are pushed directly).